ProcessLib/BoundaryCondition/HCNonAdvectiveFreeComponentFlowBoundaryCondition.cpp

```diff
@@ -0,0 +1,122 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "HCNonAdvectiveFreeComponentFlowBoundaryCondition.h"
+#include "MeshLib/Elements/Element.h"
+#include "ParameterLib/Utils.h"
+
+namespace ProcessLib
+{
+std::unique_ptr<HCNonAdvectiveFreeComponentFlowBoundaryCondition>
+createHCNonAdvectiveFreeComponentFlowBoundaryCondition(
+    BaseLib::ConfigTree const& config, MeshLib::Mesh const& bc_mesh,
+    NumLib::LocalToGlobalIndexMap const& dof_table, int const variable_id,
+    int const component_id, unsigned const integration_order,
+    std::vector<std::unique_ptr<ParameterLib::ParameterBase>> const& parameters,
+    unsigned const global_dim, Process const& process,
+    MeshLib::Mesh const& bulk_mesh, unsigned const shapefunction_order)
+{
+    DBUG(
+        "Constructing open boundary for Component Transport process from "
+        "config.");
+    //! \ogs_file_param{prj__process_variables__process_variable__boundary_conditions__boundary_condition__type}
+    config.checkConfigParameter("type",
+                                "HCNonAdvectiveFreeComponentFlowBoundary");
+
+    auto const boundary_permeability_name =
+        //! \ogs_file_param{prj__process_variables__process_variable__boundary_conditions__boundary_condition__VariableDependentNeumann__constant_name}
+        config.getConfigParameter<std::string>("parameter");
+    auto const& boundary_permeability = ParameterLib::findParameter<double>(
+        boundary_permeability_name, parameters, 1);
+    auto const bulk_node_ids =
+        bc_mesh.getProperties().template getPropertyVector<std::size_t>(
+            "bulk_node_ids", MeshLib::MeshItemType::Node, 1);
+
+    std::vector<MeshLib::Element*> const& boundary_elements =
+        bc_mesh.getElements();
+    std::vector<MeshLib::Element*> const& bulk_elements =
+        bulk_mesh.getElements();
+    std::vector<std::size_t> bulk_element_ids;
+    std::vector<std::size_t> bulk_face_ids;
+    if (!(global_dim == 3))
```

Comment:
Minor: `!(global_dim == 3)` is equivalent to `global_dim != 3` which is probably easier to read.

Reply:
Changed.